src/jit/compiler.hpp

```diff
@@ -3262,6 +3264,9 @@ unsigned           genMapIntRegNumToRegArgNum(regNumber regNum)
 inline
 unsigned           genMapFloatRegNumToRegArgNum(regNumber regNum)
 {
+#ifdef ARM_SOFTFP
+    return genMapIntRegNumToRegArgNum(regNum);
+#endif
     assert (genRegMask(regNum) & RBM_FLTARG_REGS);
```

Comment:
Rather it were the calls of genMapRegArgNumToRegNum in genFnPrologCalleeRegArgs which inside it decided which implementation to choose...